mobile/components/views/Attractions/Search/Search.tsx

```diff
@@ -0,0 +1,159 @@
+        if (userAttractionsData === attractionsExamples) {
+          Alert.alert('Error', 'Failed getting user attractions. Loading demo data');
+        } else {
+          Alert.alert('Error', 'Failed getting user attractions.');
+        }
+      })
+      .then(() => setLoading(false));
+  }, []);
+
+
+  useEffect(() => {
+    const filteredAttractions = userAttractionsData.filter(attraction => {
+      return (
+        (filters.category === "" || attraction.category === filters.category) &&
+        (filters.city === "" || attraction.city === filters.city) &&
+        (filters.country === "" || attraction.country === filters.country) &&
+        (filters.keyword === "" || attraction.name.toLowerCase().includes(filters.keyword.toLowerCase())) &&
+        (filters.region === "" || attraction.region.toLowerCase().includes(filters.region.toLowerCase()))
+      );
+    });
+    setAttractions(filteredAttractions);
+  }, [filters]);
+
+  const handleSetFilters = (filtersProps: Filters) => {
+    setFilterModalVisible(false);
+    setFilters(filtersProps);
+  };
+
+  const handleAttractionClick = (attraction: Attraction) => {
+    // setAttractionModalVisible(true);
+    setAttractionDetails(attraction);
+    navigation.navigate('AttractionDetailScreen', { id: attraction.id });
+
+    console.log('Clicked Attraction:', attraction);
+  };
+
+  const RightSwipeActions = (item: Attraction) => {
+    return (
+      <View style={{ alignItems: 'flex-end', flex: 1, marginTop: 40, padding: 10 }} >
+        <Button title='Add to favorite' onPress={() => {
+          console.log('add to favorite: ', item);
+          // swip.close();
+        }} />
+      </View>
+    );
+  };
+
+  const LeftSwipeActions = (item: Attraction) => {
+    return (
+      <View style={{ alignItems: 'flex-start', flex: 1, marginTop: 40, padding: 10 }} >
+        <Button title='Add to trip' color='green' onPress={() => console.log('add to trip: ', item)} />
+      </View>
+    );
+  };
+
+
+  return (
+    <>
+      <Button title="Filters" onPress={() => setFilterModalVisible(true)} />
+      <FiltersModal
+        visible={filterModalVisible}
+        onClose={() => setFilterModalVisible(false)}
+        onSave={handleSetFilters}
+        attractionList={userAttractionsData}
+        radiuses={radiuses}
+      />
+
+      <AttractionModal
+        visible={attractionModalVisible}
+        attraction={attractionDetails as Attraction}
+        onClose={() => setAttractionModalVisible(false)}
+      />
+
+      <View style={styles.attractionsGrid}>
+
+        {loading && (
+          <Text>Loading data</Text>
+        )}
+
+        {!loading && attractions.length === 0 && (
+          <View>
+            <Image
+              source={require('./../../../../resources/images/we_dont_have.png')}
+              style={{ width: 200, height: 200, margin: 20, padding: 10 }}
+            />
+          </View>
+        )}
+
+        {!loading && attractions.length > 0 && (
```

Comment:
You need to add here ` attractions?.length` optional chaining. The point is that this data is loaded asynchronously, if you dont add optional chaining there is high posibilty that at the moment when you will try to check `length` of `attractions`, this state will be `undefined` which will lead to causing errors

Reply:
No, because I set attractions as an empty array at the beginning
`const [attractions, setAttractions] = useState<Attraction[]>([]);`